pkg/kubelet/config/config.go

```diff
@@ -229,10 +229,18 @@ func (s *podStorage) merge(source string, change interface{}) (adds, updates, de
 				// this is a no-op
 				continue
 			}
-			// this is an add
 			recordFirstSeenTime(ref)
 			pods[name] = ref
-			adds.Pods = append(adds.Pods, ref)
+			// If a pod is not the pending phase and also not in the cache, it
```

Comment:
How is the pod cache populated after a restart? Is it based on the runtime information or API server information?